src/Flowpipes/recipes.jl

```diff
@@ -331,53 +333,21 @@ end
     seriescolor --> DEFAULT_COLOR
     seriestype --> :shape
 
-    first = true
     x = Vector{N}()
     y = Vector{N}()
-    for F in flowpipe(sol)
-    for (i, Ri) in enumerate(F)
-        if isa(F, ShiftedFlowpipe)
-            # TODO refactor; this is needed to support ShiftedFlowpipe
-            πRi = project(F, i, vars) # project the reach-set
-            Xi = set(πRi) # extract the set representation
-        else
-            Xi = _project_reachset(Ri, vars)
-        end
 
-        if Xi isa Intersection
-            res = plot_recipe(Xi, ε, Nφ)
-        else
-            # hard-code overapproximation here to avoid individual
-            # compilations for mixed sets
-            Pi = isa(Xi, AbstractPolygon) ? Xi : overapproximate(Xi, ε)
-            vlist = transpose(hcat(convex_hull(vertices_list(Pi))...))
-            if isempty(vlist)
-                @warn "overapproximation during plotting was empty"
-                continue
-            end
-            res = vlist[:, 1], vlist[:, 2]
-            # add first vertex to "close" the polygon
-            push!(res[1], vlist[1, 1])
-            push!(res[2], vlist[1, 2])
-        end
-        if isempty(res)
-            continue
-        else
-            x_new, y_new = res
-        end
-        if first
-            first = false
-        else
-            push!(x, N(NaN))
-            push!(y, N(NaN))
-        end
+    fp = flowpipe(sol)
+    for F in fp
+        x_new, y_new = _plot_reachset_list(F, N, vars, ε, Nφ)
         append!(x, x_new)
         append!(y, y_new)
     end
```

Comment:
This concatenates the vertices of different flowpipe segments. However, they should be separated by `NaN`s.

x-ref https://github.com/JuliaReach/NeuralNetworkAnalysis.jl/pull/128

![example](https://user-images.githubusercontent.com/9656686/115986418-5abb8d00-a5b0-11eb-954b-2a785b98f9b0.png)